ytmusicapi/auth/headers.py

```diff
@@ -24,13 +24,13 @@ def prepare_headers(
     session: requests.Session,
     proxies: Optional[Dict] = None,
     auth: Optional[str] = None,
+    oauth: Optional[YTMusicOAuth] = None
```

Comment:
instead of adding another parameter and making this function even more confusing, we should change the `auth` parameter to represent the `input_dict` instead of the filepath to `headers_auth`. That should greatly simplify the code. And this would also prevent reading the file twice (`load_headers_file` is called twice in this iteration)